singularity-eos/eos/eos_stiff.hpp

```diff
@@ -0,0 +1,200 @@
+    dpde = _dpde0;
+  }
+  // Generic functions provided by the base class. These contain e.g. the vector
+  // overloads that use the scalar versions declared here
+  SG_ADD_BASE_CLASS_USINGS(StiffGas)
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  static inline unsigned long scratch_size(std::string method, unsigned int nelements) {
+    return 0;
+  }
+  static inline unsigned long max_scratch_size(unsigned int nelements) { return 0; }
+  PORTABLE_INLINE_FUNCTION void PrintParams() const {
+    printf("Stiff Gas Parameters:\nGamma = %g\nCv    = %g\nPinf  = %g\nq     = %g\nqprime= %g\n", _gm1 + 1.0, _Cv,_Pinf,_qq,_qp);
+  }
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp, Real *lambda,
+                                       Real &rho, Real &sie) const {
+    sie = MYMAX(0.0, (press + (_gm1+1.0)*_Pinf)/(press + _Pinf)*_Cv*temp+_qq );
+    rho = MYMAX(0.0, (press + _Pinf) / (_gm1 * _Cv * temp) );
+  }
+  inline void Finalize() {}
+  static std::string EosType() { return std::string("StiffGas"); }
+  static std::string EosPyType() { return EosType(); }
+
+ private:
+  Real _Cv, _gm1, _Pinf, _qq, _qp;
+  // reference values
+  Real _rho0, _sie0, _bmod0, _dpde0, _entr0;
+  static constexpr const Real _T0 = ROOM_TEMPERATURE;
+  static constexpr const Real _P0 = ATMOSPHERIC_PRESSURE;
+  // static constexpr const char _eos_type[] = {"StiffGas"};
+  static constexpr const unsigned long _preferred_input =
+      thermalqs::density | thermalqs::specific_internal_energy;
+  // optional reference state variables
+};
+
+PORTABLE_INLINE_FUNCTION
+void StiffGas::FillEos(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                       Real &bmod, const unsigned long output, Real *lambda) const {
+  if (output & thermalqs::density && output & thermalqs::specific_internal_energy) {
+    if (output & thermalqs::pressure || output & thermalqs::temperature) {
+      UNDEFINED_ERROR;
+    }
+    DensityEnergyFromPressureTemperature(press, temp, lambda, rho, sie);
+  }
+  if (output & thermalqs::pressure && output & thermalqs::specific_internal_energy) {
+    if (output & thermalqs::density || output & thermalqs::temperature) {
+      UNDEFINED_ERROR;
+    }
+    sie = InternalEnergyFromDensityTemperature(rho, temp, lambda);
+  }
+  if (output & thermalqs::temperature && output & thermalqs::specific_internal_energy) {
+    sie = (press+(_gm1+1.0)*_Pinf) / (_gm1 * rho) + _qq;
+  }
+  if (output & thermalqs::pressure) press = PressureFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::temperature)
+    temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::bulk_modulus)
+    bmod = BulkModulusFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::specific_heat)
+    cv = SpecificHeatFromDensityInternalEnergy(rho, sie);
+}
```

Comment:
We haven't been the best at ensuring that `FillEos` logic is sound throughout the code. Can you make sure that there aren't any edge cases where something weird could happen here?

Reply:
I think this fills the EoS properly